File: app/src/main/java/com/yashwant/newsmodule/app/NewsApplication.java
package com.yashwant.newsmodule.app;

import android.app.Application;

import com.yashwant.newsmodule.dagger.AppComponent;
import com.yashwant.newsmodule.dagger.AppModule;
import com.yashwant.newsmodule.dagger.DaggerAppComponent;


/**
 * Created by yashwant on 16/10/17.
 */

public class NewsApplication extends Application {
    @Override
    public void onCreate() {
        super.onCreate();
        appComponent = initDagger(this);

    }

    private AppComponent appComponent;

    public AppComponent getAppComponent() {
        return appComponent;
    }

    protected AppComponent initDagger(NewsApplication application) {
        return DaggerAppComponent.builder()
                .appModule(new AppModule(application))
                .build();
    }
}

File: app/src/main/java/com/yashwant/newsmodule/app/NewsApplication.kt
package com.yashwant.newsmodule.app

import android.app.Application

import com.yashwant.newsmodule.dagger.AppComponent
import com.yashwant.newsmodule.dagger.AppModule
import com.yashwant.newsmodule.dagger.DaggerAppComponent


/**
 * Created by yashwant on 16/10/17.
 */

class NewsApplication : Application() {

    var appComponent: AppComponent? = null
        private set

    override fun onCreate() {
        super.onCreate()
        appComponent = initDagger(this)

    }

    protected fun initDagger(application: NewsApplication): AppComponent {
        return DaggerAppComponent.builder()
                .appModule(AppModule(application))
                .build()
    }
}
